assets/js/script.js

```diff
@@ -0,0 +1,270 @@
+    "video-date"
+  );
+
+  videoDetails.append(videoTitle, videoContent, videoAuthor, videoPublishedOn);
+
+  videoBox.append(videoImage, videoDetails);
+
+  return videoBox;
+};
+
+const pagination = (dataCount) => {
+  totalPage = Math.ceil(dataCount / videosPerPageCount);
+
+  const { sectionPagination } = htmlQuerySelectorObject;
+  sectionPagination.innerHTML = "";
+
+  const pagePreviousButton = createHtmlElement(
+    pageStrings.BUTTON,
+    pageStrings.PREV,
+    null,
+    pageStrings.PREVIOUS_BUTTON
+  );
+  pagePreviousButton.disabled = true;
+  setOnclickAttribute(pagePreviousButton, pageStrings.PREV);
+  sectionPagination.appendChild(pagePreviousButton);
+
+  const firstButton = createHtmlElement(
+    pageStrings.BUTTON,
+    numbers.ONE,
+    pageStrings.ACTIVE_BUTTON,
+    "btn-1"
+  );
+  setOnclickAttribute(firstButton, numbers.ONE);
+  sectionPagination.appendChild(firstButton);
+
+  for (let index = numbers.TWO; index <= totalPage; index++) {
+    const pageButton = createHtmlElement(
+      pageStrings.BUTTON,
+      index,
+      null,
+      `btn-${index}`
+    );
+    setOnclickAttribute(pageButton, index);
+    sectionPagination.appendChild(pageButton);
+  }
+
+  const pageNextButton = createHtmlElement(
+    pageStrings.BUTTON,
+    pageStrings.NEXT,
+    null,
+    pageStrings.NEXT_BUTTON
+  );
+  setOnclickAttribute(pageNextButton);
+  sectionPagination.appendChild(pageNextButton);
+  sethtmlQuerySelectorObject();
+};
+
+const paginate = async (page) => {
+  if (isCurrentPagePrevious(page)) pageStack.pop();
+  else if (isCurrentPageNext(page)) pageStack.push(++currentPage);
+  else pageStack.push(page);
+
+  currentPage = pageStack[pageStack.length - numbers.ONE];
+  setVideos();
+  toggleActiveButton();
+  toggleButton();
+};
+
+const createHtmlElement = (tagName, text, className, id) => {
+  const htmlTag = document.createElement(tagName);
+  text && (htmlTag.innerText = text);
+  id && (htmlTag.id = id);
+  className && htmlTag.classList.add(className);
+  return htmlTag;
+};
+
+const setOnclickAttribute = (button, text = pageStrings.NEXT) => {
+  button.addEventListener("click", function () {
+    paginate(
+      text == pageStrings.NEXT
+        ? pageStrings.NEXT
+        : text == pageStrings.PREV
+        ? pageStrings.PREV
+        : text,
+      this
+    );
+  });
+};
+
+const sethtmlQuerySelectorObject = () => {
+  htmlQuerySelectorObject.previousButton = document.querySelector(
+    `#${pageStrings.PREVIOUS_BUTTON}`
+  );
+  htmlQuerySelectorObject.nextButton = document.querySelector(
+    `#${pageStrings.NEXT_BUTTON}`
+  );
+  htmlQuerySelectorObject.activeButton = document.querySelector(
+    `.${pageStrings.ACTIVE_BUTTON}`
+  );
+};
+
+const toggleActiveButton = () => {
+  let { activeButton: currentActiveButton } = htmlQuerySelectorObject;
+  currentActiveButton.classList.remove(pageStrings.ACTIVE_BUTTON);
+  currentActiveButton = document.querySelector(`#btn-${currentPage}`);
```

Comment:
Please try to avoid having a different function for toggle the active/previous button. I believe we don't really need them.